src/JsonSchema/Constraints/TypeConstraint.php

```diff
@@ -249,14 +267,46 @@ protected function toNumber($value)
         if (is_numeric($value)) {
             return $value + 0; // cast to number
         }
+        if (is_bool($value) || is_null($value)) {
+            return (int) $value;
+        }
+        if ($this->getTypeCheck()->isArray($value) && count($value) === 1) {
+            reset($value);
+
+            return $this->toNumber(current($value));
+        }
 
         return $value;
     }
 
+    /**
+     * Converts a value to an integer
+     *
+     * @param mixed $value
+     *
+     * @return mixed
+     */
     protected function toInteger($value)
     {
-        if (is_numeric($value) && (int) $value == $value) {
-            return (int) $value; // cast to number
+        $numberValue = $this->toNumber($value);
```

Comment:
that's what I mean, though--if you put the code back the way it was then there is only one call to `is_numeric`. So I guess I'm still not clear on what is functionally different about the new code. Are additional values now being coerced to integer (eg. boolean) with this code?